packages/devtools_app/lib/src/extensions/extension_settings.dart

```diff
@@ -0,0 +1,204 @@
+}
+
+class __ExtensionsListState extends State<_ExtensionsList> {
+  late ScrollController scrollController;
+
+  late final List<DevToolsExtensionConfig> orderedExtensions;
+
+  @override
+  void initState() {
+    super.initState();
+    orderedExtensions = List<DevToolsExtensionConfig>.of(widget.extensions)
+      ..sort((a, b) => a.name.compareTo(b.name));
+    scrollController = ScrollController();
+  }
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scrollbar(
+      controller: scrollController,
+      thumbVisibility: true,
+      child: ListView.builder(
+        controller: scrollController,
+        itemCount: orderedExtensions.length,
+        itemBuilder: (context, index) => ExtensionSetting(
+          extension: orderedExtensions[index],
+        ),
+      ),
+    );
+  }
+}
+
+@visibleForTesting
+class ExtensionSetting extends StatelessWidget {
+  const ExtensionSetting({super.key, required this.extension});
+
+  final DevToolsExtensionConfig extension;
+
+  @override
+  Widget build(BuildContext context) {
+    final states = [
+      (
+        'Enabled',
+        (state) => state == ExtensionEnabledState.enabled,
+        () {
+          ga.select(
+            gac.extensionSettingsId,
+            gac.extensionEnable(extension.name.toLowerCase()),
+          );
+          unawaited(
+            extensionService.setExtensionEnabledState(
+              extension,
+              enable: true,
+            ),
+          );
+        },
+      ),
+      (
+        'Disabled',
+        (state) => state == ExtensionEnabledState.disabled,
+        () {
+          ga.select(
+            gac.extensionSettingsId,
+            gac.extensionDisable(extension.name.toLowerCase()),
+          );
+          unawaited(
+            extensionService.setExtensionEnabledState(
+              extension,
+              enable: false,
+            ),
+          );
+        },
+      ),
+    ];
+    final theme = Theme.of(context);
+    final extensionName = extension.name.toLowerCase();
+    return ValueListenableBuilder(
+      valueListenable: extensionService.enabledStateListenable(extensionName),
+      builder: (context, enabledState, _) {
+        return Padding(
+          padding: const EdgeInsets.only(bottom: denseSpacing),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text(
+                'package:$extensionName',
+                overflow: TextOverflow.ellipsis,
+              ),
+              DevToolsToggleButtonGroup(
+                fillColor: theme.colorScheme.primary,
+                selectedColor: theme.colorScheme.onPrimary,
+                onPressed: (index) => states[index].$3(),
+                selectedStates:
+                    states.map((option) => option.$2(enabledState)).toList(),
+                children: states
```

Comment:
This was making the analysis server unhappy:
![Screenshot 2023-08-01 at 2 44 10 PM](https://github.com/flutter/devtools/assets/43759233/0da00f72-921f-4baa-b7a8-3f5bf329cb4f)

For some reason the order of the items in the record were not in the order that the analyzer expected. Could have been coincidental but it expected them in alphabetical order, not the order in which they were defined.

Reply:
Ah, this would have worked with the positional records.

This should work now that we're using named properties:

```dart
selectedStates: [
  for (final state in states)
    state.isSelected(enabledState),
],
children: [
  for (final state in states)
    Padding(
      padding: const EdgeInsets.symmetric(
        vertical: densePadding,
        horizontal: denseSpacing,
      ),
      child: Text(state.name),
    ),
]
```